### DISCUSSION TITLE
How to refresh account balances

### DISCUSSION THREAD

**Discussion by mbasith** [hammem/monarchmoney#42] [Q&A] [answered]
Is there a method to refresh all account balances (ForceRefreshAccountsQuery)?

**Comment by CalvinChanCan** [comment, marked as answer]
Hi @mbasith . There currently isn't a method to trigger the account balance refresh. Though taking a look at the API, it does seem possible through the GraphQL query you mentioned. 

I added this as a feature request here and when I have some free time, I'll give this a try. Th
https://github.com/hammem/monarchmoney/issues/43

Though what kind of response from this method should this return? It seems like when forcing the account refresh, the response is just whether it was successful in triggering the force refresh and not when the forced refresh is completed. Should this wait until the account refresh is completed? Or maybe we could include another method to periodically check whether all accounts have completed the refresh similar to the Web UI? Thoughts on this @hammem ?

**Reply by hammem** [reply]
I'm with you @CalvinChanCan , definitely a good thing to add.  I added some comments on #43 about implementation.

**Comment by emann3** [comment]
where is the documentation for the monarch money api? I have not been able to find it thru Google,  and would like to see what other commands are available.

Thanks

**Reply by grablair** [reply]
AFAIK it isn't public, though others can feel free to correct me. I implemented the `create_transaction` feature via reverse engineering (inspecting the network calls when I attempt to perform a given action on the website).

**Reply by hammem** [reply]
+1 I've been using the web UX with Chrome devtools to determine the GraphQL calls needed.
